In [3]:
import matplotlib.pyplot as plt
import pandas as pd
import numpy as np
import openpyxl

print('Teste de funcionamento e versão de bibliotecas')
print("Pandas:", pd.__version__)
print("Matplotlib:", plt.matplotlib.__version__)
print("NumPy:", np.__version__)
print("OpenPyXL:", openpyxl.__version__)

Teste de funcionamento e versão de bibliotecas
Pandas: 3.0.3
Matplotlib: 3.10.9
NumPy: 2.4.6
OpenPyXL: 3.1.5


In [4]:
# Informar caminho do arquivo Excel
arquivo_dataset = '../dados/analise_logistica_dataset.xlsx'

In [5]:
excel = pd.ExcelFile(arquivo_dataset)
excel.sheet_names

['PRODUTOS', 'CLIENTES', 'PEDIDOS', 'ENTREGAS']

In [6]:
produtos = pd.read_excel(arquivo_dataset, sheet_name='PRODUTOS')

In [7]:
produtos.head()

,produto_id,produto_nome,categoria,fornecedor,peso_kg,preco_unitario
0,P001,Notebook Pro 14,Computadores,Tech Solutions,1.8,4500
1,P002,Notebook Ultra 15,Computadores,Digital Prime,2.1,5200
2,P003,Monitor LED 24,Monitores,Tech Solutions,3.2,850
3,P004,Monitor Gamer 27,Monitores,Gamer Store,5.0,1800
4,P005,Teclado Mecânico RGB,Periféricos,Gamer Store,0.9,350


In [8]:
produtos.shape

(30, 6)

In [9]:
produtos.dtypes

produto_id            str
produto_nome          str
categoria             str
fornecedor            str
peso_kg           float64
preco_unitario      int64
dtype: object

In [10]:
produtos.isnull().sum()

produto_id        0
produto_nome      0
categoria         0
fornecedor        0
peso_kg           0
preco_unitario    0
dtype: int64

In [11]:
produtos.duplicated().sum()

np.int64(0)

In [12]:
produtos['produto_id'].nunique()

30

In [13]:
clientes = pd.read_excel(arquivo_dataset, sheet_name='CLIENTES')

In [14]:
clientes.head()

,cliente_id,empresa_nome,cidade,estado,regiao
0,C001,TecNova Soluções,Recife,PE,Nordeste
1,C002,Alpha Sistemas,João Pessoa,PB,Nordeste
2,C003,Connect Tecnologia,Fortaleza,CE,Nordeste
3,C004,Inova Digital,Salvador,BA,Nordeste
4,C005,Smart Business,Maceió,AL,Nordeste


In [15]:
clientes.shape

(50, 5)

In [16]:
clientes.dtypes

cliente_id      str
empresa_nome    str
cidade          str
estado          str
regiao          str
dtype: object

In [17]:
clientes.isnull().sum()

cliente_id      0
empresa_nome    0
cidade          0
estado          0
regiao          0
dtype: int64

In [18]:
clientes.duplicated().sum()

np.int64(0)

In [19]:
clientes['cliente_id'].nunique()

50

In [21]:
pedidos = pd.read_excel(arquivo_dataset, sheet_name='PEDIDOS')

In [22]:
pedidos.shape

(100, 6)

In [23]:
pedidos.dtypes

pedido_id                 str
cliente_id                str
produto_id                str
data_pedido    datetime64[us]
quantidade              int64
valor_total             int64
dtype: object

In [24]:
pedidos.isnull().sum()

pedido_id      0
cliente_id     0
produto_id     0
data_pedido    0
quantidade     0
valor_total    0
dtype: int64

In [25]:
pedidos.duplicated().sum()

np.int64(0)

In [26]:
pedidos['pedido_id'].nunique()

100

In [27]:
entregas = pd.read_excel(arquivo_dataset, sheet_name='ENTREGAS')

In [28]:
entregas.shape

(100, 7)

In [29]:
entregas.dtypes

entrega_id                   str
pedido_id                    str
transportadora               str
data_envio        datetime64[us]
data_entrega      datetime64[us]
status_entrega               str
prazo_dias                 int64
dtype: object

In [30]:
entregas.isnull().sum()

entrega_id        0
pedido_id         0
transportadora    0
data_envio        0
data_entrega      0
status_entrega    0
prazo_dias        0
dtype: int64

In [31]:
entregas.duplicated().sum()

np.int64(0)

In [32]:
entregas['entrega_id'].nunique()

100

In [33]:
pedidos['cliente_id'].isin(clientes['cliente_id']).all()

np.True_

In [34]:
pedidos['produto_id'].isin(produtos['produto_id']).all()

np.True_

In [35]:
entregas['pedido_id'].isin(pedidos['pedido_id']).all()

np.True_

In [36]:
pedidos['valor_total'].sum()

np.int64(197670)

In [37]:
pedidos['pedido_id'].nunique()

100

In [38]:
pedidos['valor_total'].mean()

np.float64(1976.7)

In [39]:
pedidos.groupby('produto_id')['quantidade'].sum().sort_values(ascending=False)

produto_id
P020    42
P006    26
P007    21
P005    19
P026    18
P022    15
P028    12
P009    11
P017    11
P018    11
P003    10
P029     9
P019     8
P030     8
P011     6
P023     5
P021     5
P012     5
P008     5
P001     5
P002     5
P014     5
P027     5
P024     4
P016     4
P004     3
P025     3
P013     3
P010     2
P015     2
Name: quantidade, dtype: int64

In [40]:
vendas_produtos = pedidos.merge(
    produtos[['produto_id', 'produto_nome']],
    on='produto_id'
)

vendas_produtos.groupby('produto_nome')['quantidade'].sum().sort_values(ascending=False)

produto_nome
Pen Drive 128GB              42
Mouse Gamer RGB              26
Mouse Sem Fio                21
Teclado Mecânico RGB         19
Cabo HDMI 2 Metros           18
Memória RAM 16GB             15
Power Bank 20000mAh          12
Fone Bluetooth               11
Roteador Wi-Fi 6             11
SSD Externo 1TB              11
Monitor LED 24               10
Luminária LED Inteligente     9
HD Externo 2TB                8
Câmera de Segurança Wi-Fi     8
Smartphone X10                6
Placa de Vídeo RTX            5
Notebook Pro 14               5
Notebook Ultra 15             5
Smartphone Pro Max            5
Processador Gamer             5
Carregador USB-C              5
Headset Gamer                 5
Smartwatch Fit                5
Cadeira Gamer                 4
Impressora Multifuncional     4
Mesa para Computador          3
Monitor Gamer 27              3
Tablet 10 Polegadas           3
Caixa de Som Bluetooth        2
Câmera Web HD                 2
Name: quantidade, dtype: in

In [41]:
vendas_por_produto = (
    vendas_produtos
    .groupby('produto_nome')['quantidade']
    .sum()
    .sort_values(ascending=False)
)

In [42]:
vendas_por_produto.head(5)

produto_nome
Pen Drive 128GB         42
Mouse Gamer RGB         26
Mouse Sem Fio           21
Teclado Mecânico RGB    19
Cabo HDMI 2 Metros      18
Name: quantidade, dtype: int64

In [43]:
vendas_produtos.groupby('produto_nome')['valor_total'].sum().sort_values(ascending=False).head(5)

produto_nome
Notebook Ultra 15     26000
Notebook Pro 14       22500
Smartphone Pro Max    22500
Placa de Vídeo RTX    17500
Smartphone X10        16800
Name: valor_total, dtype: int64

In [44]:
pedidos_clientes = pedidos.merge(
    clientes[['cliente_id', 'regiao']],
    on='cliente_id'
)

In [45]:
pedidos_clientes.groupby('regiao')['valor_total'].sum().sort_values(ascending=False)

regiao
Sudeste         59850
Sul             42560
Centro-Oeste    37620
Norte           31340
Nordeste        26300
Name: valor_total, dtype: int64

In [46]:
entregas['status_entrega'].value_counts()

status_entrega
Entregue    69
Atrasada    31
Name: count, dtype: int64

## Principais Insights

- O faturamento total dos 100 pedidos foi de **R$ 197.670,00**, com um ticket médio de **R$ 1.976,70 por pedido**.

- O **Pen Drive 128GB** foi o produto com maior quantidade de unidades vendidas, totalizando **42 unidades**.

- A região **Sudeste** apresentou o maior faturamento, com **R$ 59.850,00**, enquanto o **Nordeste** apresentou o menor, com **R$ 26.300,00**.

- Do total de 100 entregas analisadas, **69 foram entregues** e **31 foram classificadas como atrasadas**, representando **31% de entregas atrasadas**.

- Os produtos com maior quantidade vendida não são necessariamente os que apresentam maior faturamento, devido às diferenças de preço entre os produtos.